libs/ui-lib/lib/ocm/components/clusterConfiguration/staticIp/components/FormViewHosts/FormViewHostsFields.tsx

```diff
@@ -1,38 +1,119 @@
+          setFieldValue(`${fieldName}.bondPrimaryInterface`, '');
+          setFieldValue(`${fieldName}.bondSecondaryInterface`, '');
+        }
+      } else {
+        setUseBond(true);
+      }
+    };
+
+    const handleModalConfirm = () => {
+      setUseBond(false);
+      setFieldValue(`${fieldName}.bondType`, 'active-backup');
+      setFieldValue(`${fieldName}.bondPrimaryInterface`, '');
+      setFieldValue(`${fieldName}.bondSecondaryInterface`, '');
+      setIsModalOpen(false);
+    };
+
+    const handleModalCancel = () => {
+      setIsModalOpen(false);
+    };
     return (
-      <Grid hasGutter>
-        <OcmInputField
-          name={`${fieldName}.macAddress`}
-          label="MAC Address"
-          isRequired
-          data-testid={`mac-address-${hostIdx}`}
-        />
-        {getShownProtocolVersions(protocolType).map((protocolVersion) => (
-          <FormGroup
-            label={`IP address (${getProtocolVersionLabel(protocolVersion)})`}
-            fieldId={getFieldId(`${fieldName}.ips.${protocolVersion}`, 'input')}
-            key={protocolVersion}
-          >
-            <OcmInputField
-              name={`${fieldName}.ips.${protocolVersion}`}
-              isRequired
-              data-testid={`${protocolVersion}-address-${hostIdx}`}
-            />{' '}
+      <>
+        <Grid hasGutter>
+          <FormGroup>
+            <OcmCheckboxField
+              label={
+                <>
+                  {'Use bond'}{' '}
+                  <PopoverIcon
+                    noVerticalAlign
+                    bodyContent="Bonds help you to combine network interfaces  for increased bandwidth and ensure redundancy. To bond more than 2 network interfaces per host, use the YAML view."
+                  />
+                </>
+              }
+              onChange={(value) => handleUseBondChange(value)}
+              name={`${fieldName}.useBond`}
+            />
           </FormGroup>
-        ))}
-      </Grid>
+          {useBond && (
+            <Grid hasGutter className="pf-v5-u-ml-lg">
+              <FormGroup fieldId={`bond-type-${hostIdx}`}>
+                <BondsSelect name={`${fieldName}.bondType`} data-testid={`bond-type-${hostIdx}`} />
+              </FormGroup>
+              <OcmInputField
+                name={`${fieldName}.bondPrimaryInterface`}
+                label="Port 1 MAC Address"
+                data-testid={`bond-primary-interface-${hostIdx}`}
+                isRequired
+              />{' '}
+              <OcmInputField
+                name={`${fieldName}.bondSecondaryInterface`}
+                label="Port 2 MAC Adddress"
+                data-testid={`bond-secondary-interface-${hostIdx}`}
+                isRequired
+              />
+            </Grid>
+          )}
+          <OcmInputField
+            name={`${fieldName}.macAddress`}
+            label="MAC Address"
+            isRequired
+            data-testid={`mac-address-${hostIdx}`}
+          />
```

Comment:
And this needs to be gone if the users select to use bonds, I think. (And validations need to reflect that as well.)

Reply:
Asked in slack thread: https://redhat-internal.slack.com/archives/C07TXR13N66/p1730283814234659

Reply:
When use bond is check we shouldn't have the extra mac-address field.